### ISSUE TITLE
Incorrect LaTeX highlight with half-open intervals and macros

### ISSUE THREAD

**Issue by cionx** [nvim-treesitter/nvim-treesitter#1285]
**Describe the bug**

The syntax highlight for LaTeX doesn’t work correctly with half-open intervals and (certain?) macros.

**To Reproduce**

Consider the following LaTeX code:

<details>

```latex
\documentclass{scrartcl}

\usepackage{mathtools}

\begin{document}

	Let $A \coloneqq (-\infty, 0]$ and $B \coloneqq [0, \infty)$.

\end{document}
```

</details>
The resulting syntax highlight looks as follows:

![output](https://user-images.githubusercontent.com/7231796/117585058-af164f00-b110-11eb-8d32-8202555b6e50.png)

**Expected behavior**

In the piece of code `$ and $`, the word `and`  should be treated as regular text, i.e. in the same way as `Let` at the beginning of the line.

**Output of `:checkhealth nvim_treesitter`**

<details>
<code>

health#nvim_treesitter#check
========================================================================
## Installation
  - OK: `tree-sitter` found  0.19.4 (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v16.1.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl" }
  - OK: Neovim was compiled with tree-sitter runtime ABI version 13 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I
  - latex          ✓ . ✓ . 
  - c              ✓ ✓ ✓ ✓ 
  - python         ✓ ✓ ✓ ✓ 

 Legend: H[ighlight], L[ocals], F[olds], I[ndents]
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}

</code>
</details>

**Output of `nvim --version`**

<details>
<code>

NVIM v0.5.0-dev+1311-gf8173df4d
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/runner/work/neovim/neovim/build/config -I/home/runner/work/neovim/neovim/src -I/home/runner/work/neovim/neovim/.deps/usr/include -I/usr/include -I/home/runner/work/neovim/neovim/build/src/nvim/auto -I/home/runner/work/neovim/neovim/build/include
Compiled by runner@fv-az50-511

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/share/nvim"

Run :checkhealth for more info

</code>
</details>

**Additional context**

The problem does not occur under the following circumstances:
- `( )` or `[ ]` instead of `( ]` and `[ )`.
- `=` or `≔` instead of `\coloneqq`.
- `[ [` instead of `[ )` and `] ]` instead of `( ]`. (But then other things break.)

**Comment by theHamsta** [comment]
This seems to be a issue with https://github.com/latex-lsp/tree-sitter-latex. Can you report the problem there?

@pfoerster

**Comment by cionx** [comment]
The problem seems to have been fixed with latex-lsp/tree-sitter-latex@7f720661de5316c0f8fee956526d4002fa1086d8. I also tested the syntax highlight by changing the revision in `lockfile.json` and using `TSUpdate`, and it now looks as expected.

![Screenshot_20210511_191300](https://user-images.githubusercontent.com/7231796/117857945-a4cd8f80-b28d-11eb-9368-4504190453e5.png)
